cli/cppcheckexecutor.cpp

```diff
@@ -158,7 +159,40 @@ bool CppCheckExecutor::parseFromArgs(CppCheck *cppcheck, int argc, const char* c
 #else
     const bool caseSensitive = true;
 #endif
-    if (!pathnames.empty()) {
+    if (!mSettings->project.fileSettings.empty()) {
+        // filter only for the selected filenames from all project files
+        std::list<ImportProject::FileSettings> newList;
+
+        for (const std::string &fname : pathnames) {
+            std::string searchString = fname;
+            size_t pos = fname.find('*');
+
+            // if the user added an asterix '*' replace it with regex '\w+' to match whole words
```

Comment:
In suppressions, a '*' means any characters as far as I remember.

I really think that inconsistent. If you execute cppcheck like this:

    cppcheck file1.cpp

then cppcheck will not check the file `foo/bar/file1.cpp`

If you execute cppcheck like this:

    cppcheck --project=foobar.vcxproj file1.cpp

Then suddenly you want that we check the file `foo/bar/file1.cpp`.

Also imagine that the user writes:

    cppcheck --project=foobar.vcxproj a

Then as far as I see, any path that contains a "a" will be checked:

    foo/bar/file1.cpp

If I would write "a" then I would probably only want that files in the "a" folder would be checked.

Please break out this pattern matching into a function so it can be unit tested.